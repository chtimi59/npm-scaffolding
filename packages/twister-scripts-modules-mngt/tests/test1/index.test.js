const path = require('path')
const childProcess = require('node-child-process-extension')
const fs = require('node-fs-extension')
const nodePrefix = `node ${path.resolve(__dirname, '../../../npm-twister/npmt.js')}`
async function npmt(args="") {
    cmd = `${nodePrefix} ${args}`
    return childProcess.extras.exeToStdOut(cmd, __dirname)
}
const packageJson = path.resolve(__dirname, 'package.json')
const nodeModules = path.resolve(__dirname, 'node_modules')

describe('twister-scripts-modules-mngt', function () {
    describe('test1', function () {
        beforeAll(async function () {
            await fs.extras.rm(nodeModules)
            await fs.extras.writeJsonSync(packageJson, {
                "extends": "a/alt.js",
                "name": "mgnt-test1",
                "version": "1.0.0"
            })
        }, 15000)
        it('hello chalk', async function () {
            await npmt("hello chalk")
        }, 15000)
        it('install', async function () {
            await npmt("hello")
            // "color-name": "rm"
            expect(await fs.extras.exists(path.resolve(nodeModules, "color-name"), "folder"))
                .toEqual(false)
            // "chalk-bis": "rename chalk",
            expect(await fs.extras.exists(path.resolve(nodeModules, "chalk"), "folder"))
                .toEqual(false)
            expect(await fs.extras.exists(path.resolve(nodeModules, "chalk-bis"), "folder"))
                .toEqual(true)
            // "new-foo": "cp ./foo",
            expect(fs.readFileSync(path.resolve(nodeModules, "new-foo", "index.js")).toString())
                .toEqual("console.log('new foo')")
            // "src-foo": "symlink ./foo",
            expect(fs.readFileSync(path.resolve(nodeModules, "src-foo", "index.js")).toString())
                .toEqual("console.log('new foo')")
            // "color-convert-2": "symlink ../node_modules/color-convert"
            expect(await fs.extras.exists(path.resolve(nodeModules, "color-convert-2"), "folder"))
                .toEqual(true)
        }, 15000)
        it('re-install', async function () {
            const ret = await npmt("hello")
            expect(ret).toEqual("")
        }, 15000)
    })
})
