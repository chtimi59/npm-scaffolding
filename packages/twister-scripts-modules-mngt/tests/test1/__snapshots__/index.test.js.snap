// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`twister-scripts-modules-mngt test1 empty 1`] = `
"Usage:  npmt <command>

wrapped commands from npm (v6.4.1):

    available:
    access, add-user, adduser, audit, author, bin, bugs, c, cache, ci
    completion, config, create, ddp, dedupe, deprecate, dist-tag, dist-tags, docs, doctor
    edit, explore, find, find-dupes, get, help, help-search, home, hook, info
    init, innit, install, issues, it, la, link, list, ll, ln
    login, logout, ls, outdated, owner, pack, ping, prefix, profile, prune
    publish, r, rb, rebuild, remove, repo, restart, rm, root, rum
    run, run-script, s, se, search, set, show, shrinkwrap, star, stars
    start, stop, t, team, test, token, tst, udpate, un, uninstall
    unlink, unpublish, unstar, up, update, upgrade, urn, v, version, view
    whoami

    override:
    add, i, isntall


where:

    hello       install prerequist
                alias: i, isntall, add

Specific help example:

    npmt install --help

version: 0.0.1"
`;
